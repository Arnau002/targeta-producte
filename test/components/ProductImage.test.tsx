import React from 'react';
import renderer from 'react-test-renderer';

import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('Tests <ProductImage />', () => { 
    test('Mostra el component amb la imatge personalitzada', () => { 
                
        const wrapper = renderer.create(
            <ProductImage img='https://test.png' />
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
    });
    

    test('Mostra el component amb la imatge del producte', () => { 
        const wrapper = renderer.create(
            <ProductCard product={ product2 }>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
            
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
    });
 });