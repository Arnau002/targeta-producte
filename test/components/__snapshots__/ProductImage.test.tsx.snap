// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests <ProductImage /> Mostra el component amb la imatge del producte 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Tests <ProductImage /> Mostra el component amb la imatge personalitzada 1`] = `
<img
  alt="Product image"
  className="productImg undefined"
  src="https://test.png"
/>
`;
